### PR TITLE
Some mechanical alterations to make text better-looking

### PR DESCRIPTION
**Pull request by rsperberg**
These changes make the text more readable and less typewriter-like by using curly quotes and em dashes.

### CODE REVIEW

**Comment by ericelliott** [comment]
This is a fine idea, but it's also a maintenance nightmare. What we really need is a build script that will automate this process. Is that something you'd be willing to take on? =)

**Comment by rsperberg** [comment]
I agree that these types of changes should be automated (and that they are, for the most part, automatable). I know what regexes to write.

I'm willing.

However, you'll probably have to tell me just what you want done and maybe stand over my shoulder as I get started. I'm that new to this sort of thing. :-|

**Comment by ericelliott** [comment]
For starters, a simple node script that takes a string as an input and returns the transformed results as output. I'm happy starting with apostrophes, quotes, and em dashes.

Quick question: How do you distinguish between an apostrophe and single-quotes? Does this matter?

**Comment by rsperberg** [comment]
Most apostrophes are both preceded and followed by letters (and digits, e.g., 1980’s). Single quotes usually have spaces or line returns preceding or following them.

The apostrophe and the right single quote are the same glyph so distinguishing them doesn't matter from a typesetting standpoint.

I'll take a hack at a first-draft tomorrow.

**Comment by ericelliott** [comment]
Sounds great. =)

**Comment by rsperberg** [comment]
I've committed a basic script that handles most of the apostrophe, quote and em-dash replacements.

There are a few replacement rules I haven't gotten to yet.

And structurally, the script leaves a lot to be desired. I think I can see clear to rewriting it so that it makes use of `Array.forEach()` over a single array of objects that pair the search regexes and replacement strings. 

Also, I'm not sure how you want to structure the site, but I figured `/assets/js` was neutral enough as a start-point.

**Comment by ericelliott** [comment]
Good start. I'd move that into a `scripts` directory instead of `assets/js`, and make sure there's a script entry in `package.json` to call it.

Of course, that hard-coded file path isn't going to work on anybody's computer but yours. You should probably just read the input from process.stdin and write the output to process.stdout. The package.js entry could do:

```
"scripts": {
  "cleanup": "cat README.md | node scripts/cleanup.js > build/README.md"
},
```

**Comment by rsperberg** [comment]
I've moved the js file to the scripts directory. I haven't gotten to the issue of the hard-coded path.

However, I have rewritten the mechanics of replacing text so that it uses a single array.

**Comment by rsperberg** [comment]
I've changed the script to run as suggested (well, not exactly as I didn't use process.stdin) and added a package.json file.

Works for me as intended when entering `npm run-script cleanup` from the command line

**Comment by ericelliott** [comment]
Please merge my latest changes into this branch so we can merge this. =)

**Comment by ericelliott** [comment]
Also, please don't alter the original markdown file. You can copy and paste the current version of that to replace the one in this branch. The only Markdown changes should go into the new build directory.

**Comment by rsperberg** [comment]
OK, won't change that md file.

Let me know if I didn't get this right.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/misc
```
Comment (ericelliott) [inline .gitignore]: What is this?

Comment (rsperberg) [inline .gitignore]: Ah, I put some notes to myself in a miscellaneous folder (/misc)and then figured GitHub needed to be told to pay no attention to what I put in that folder. Is there some way to keep changes I make to .gitignore from showing up as part of a pull request?

Comment (rsperberg) [inline .gitignore]: Looking at it, I can't figure what led that slash to appear at the beginning rather than the end of the folder name.

Comment (ericelliott) [inline .gitignore]: For now you should delete it and save your personal outside the git repo. =)

**Comment by rsperberg** [comment]
Missed your note. Have moved `misc/` outside the git repo and deleted the `.gitignore` since it had only a reference to that folder.

**Comment by ericelliott** [comment]
Nice work! :+1:

**Pull request merged**